config/webhook/webhook.go

```diff
@@ -33,15 +42,79 @@ func main() {
 }
 
 type EvictionHandler struct {
+	Client  client.Client
 	decoder *admission.Decoder
 }
 
 func (e *EvictionHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
 	logger.Info("Received eviction request", "namespace", req.Namespace, "name", req.Name)
 
 	// Log eviction request
-	fmt.Printf("Eviction requested for Pod %s/%s\n", req.Namespace, req.Name)
+	evictionLog := myappsv1.EvictionLog{
+		PodName:      req.Name,
+		EvictionTime: time.Now().Format(time.RFC3339),
+	}
+
+	// Fetch the pod to get its labels
+	pod := &corev1.Pod{}
+	err := e.Client.Get(ctx, types.NamespacedName{Namespace: req.Namespace, Name: req.Name}, pod)
+	if err != nil {
+		logger.Error(err, "Unable to fetch Pod")
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	// List all PDBWatchers in the namespace
+	pdbWatcherList := &myappsv1.PDBWatcherList{}
+	err = e.Client.List(ctx, pdbWatcherList, &client.ListOptions{Namespace: req.Namespace})
+	if err != nil {
+		logger.Error(err, "Unable to list PDBWatchers")
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	// Find the applicable PDBWatcher
+	var applicablePDBWatcher *myappsv1.PDBWatcher
+	for _, pdbWatcher := range pdbWatcherList.Items {
+		// Fetch the associated PDB
+		pdb := &policyv1.PodDisruptionBudget{}
+		err := e.Client.Get(ctx, types.NamespacedName{Name: pdbWatcher.Spec.PDBName, Namespace: pdbWatcher.Namespace}, pdb)
+		if err != nil {
+			logger.Error(err, "Unable to fetch PDB", "pdbName", pdbWatcher.Spec.PDBName)
+			continue
+		}
+
+		// Check if the PDB selector matches the evicted pod's labels
+		selector, err := metav1.LabelSelectorAsSelector(pdb.Spec.Selector)
+		if err != nil {
+			logger.Error(err, "Invalid PDB selector", "pdbName", pdbWatcher.Spec.PDBName)
+			continue
+		}
+
+		if selector.Matches(labels.Set(pod.Labels)) {
+			applicablePDBWatcher = &pdbWatcher
+			break
```

Comment:
Do we have something else in controller to error if two pdb watchers point at same pdb?